Tennis-scheduler/src/main/java/com/tennis/tennisscheduler/controllers/TimeslotController.java

```diff
@@ -89,4 +104,25 @@ public ResponseEntity<HttpStatus> deleteById(@PathVariable long id){
         timeslotService.deleteById(id);
         return new ResponseEntity<>(HttpStatus.OK);
     }
+    @PreAuthorize("hasAnyRole('ADMIN','TENNIS_PLAYER')")
+    @PutMapping(value = "/cancel/{id}")
+    public ResponseEntity<HttpStatus> cancelTimeslot(@PathVariable long id){
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Person user = (Person)authentication.getPrincipal();
+
+        Timeslot timeslotExisting = timeslotService.getById(id);
+        if (timeslotExisting == null) {
+            return new ResponseEntity<>( HttpStatus.BAD_REQUEST);
+        }
+
+        if(user.getRole().getRoleName().equals(UserType.ROLE_TENNIS_PLAYER)){
```

Comment:
Check with Danilo regarding this, he had a similar thing implemented for updatePerson method. It can be written a lot shorter and clearer.